pallets/gear/src/lib.rs

```diff
@@ -874,10 +962,23 @@ pub mod pallet {
                 }
             }
 
+            let post_data: QueuePostProcessingData = ext_manager.into();
+
+            // TODO: consider about merging this data into separate event or `Event::MessagesDispatched`
+            for id in post_data.state_changes {
```

Comment:
This is the place I do care about: if there would be processed 100 programs then about 100 events would be emitted. So  I agree with what said in the comment: merge all data in the one event.

Reply:
`CodeChanged` can't be interpreted as `CodeChanged` due to the fact of processing especially in extrinsic and a bit cases in process_queue in future. Same here about initialising programs - they took place in extrinsics and in mq processing also. What's the best option how to solve this? I really would like to open issue and consider in future

The only easy option I see is to separate life-cycle dependent stuff about programs from their state changes data. How do you feel about it?